hazelcast/src/main/java/com/hazelcast/cache/impl/operation/PostJoinCacheOperation.java

```diff
@@ -38,14 +43,31 @@ public void addCacheConfig(CacheConfig cacheConfig) {
 
     @Override
     public String getServiceName() {
-        return ICacheService.SERVICE_NAME;
+        return SERVICE_NAME;
     }
 
     @Override
     public void run() throws Exception {
-        ICacheService cacheService = getService();
-        for (CacheConfig cacheConfig : configs) {
-            cacheService.putCacheConfigIfAbsent(cacheConfig);
+        if (isJcacheAvailable(getNodeEngine().getConfigClassLoader())) {
```

Comment:
Thanks @ahmetmircik , I renamed the methods in `JCacheDetector` with a capital C in a separate commit. Once green will squash & merge.